FindMe/Model/Contact.m

```diff
@@ -7,17 +7,39 @@
 //
 
 #import "Contact.h"
+@import Contacts;
+@import ContactsUI;
 
 @implementation Contact
 
-    @dynamic telephoneNumber;
-    @dynamic name;
-    @dynamic user;
-    @dynamic email;
+@dynamic telephoneNumber;
+@dynamic name;
+@dynamic user;
+@dynamic email;
+@dynamic lastName;
 
 + (nonnull NSString *)parseClassName
 {
     return @"Contact";
 }
 
++ (void)uploadContacts: (NSArray<CNContact*> *)contacts withCompletion: (PFBooleanResultBlock  _Nullable)completion
+{
+    if(contacts) {
+        for (CNContact *contact in contacts) {
+            Contact *newContact = [Contact new];
+            newContact.name = contact.givenName;
+            newContact.user = [PFUser currentUser];
+            if(contact.givenName) {
+                newContact.lastName = contact.familyName;
+            }
+            if (contact.emailAddresses) {
+                newContact.email = contact.emailAddresses.firstObject.value;
+            }
+            newContact.telephoneNumber = contact.phoneNumbers.firstObject.value.stringValue;
```

Comment:
Could do more validation here, such as whether it's all digits or not.

Reply:
Could this be the case if it's an international phone number with, for example, a +1 at the start?